models/image.py

```diff
@@ -161,11 +157,16 @@ def new_image(self, value):
             raise ValueError('Do not assign None to new_image. Simply clear the upstream_images list.')
         if not isinstance(value, Image):
             raise ValueError("The new_image must be an Image object.")
-        if len(self.upstream_images) != 2:
-            raise ValueError("This only works for subtractions with exactly two upstream images.")
+        if len(self.upstream_images) not in [1, 2]:
+            raise ValueError("This only works for subtractions that have one or two upstream images.")
```

Comment:
How can a subtraction only have one upstream image?

Reply:
This happens when you are building up the object. If you want to assign the ref and new one after the other, you'd run into a step where it already has a reference but doesn't have the new image yet. This lets you assign the new image.